File: 139-word-break/word-break.java
class Solution {
    public boolean wordBreak(String s, List<String> wordDict) {
        int len_dp = s.length() + 1;
        boolean dp [] = new boolean[len_dp];
        dp[0] = true;

        List<Integer> trues = new ArrayList<>();
        trues.add(0);

        for(int i=1; i<len_dp; i++){
            for (int j : trues){
               if(wordDict.contains(s.substring(j,i))){
                dp[i] = true;
                trues.add(i);
                break;
               } 
            }
        }
        return dp[len_dp - 1];
    }
}

File: 139-word-break/word-break.py
class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        len_dp = len(s) + 1
        dp = [False] * len_dp
        dp[0] = True
        trues = [0] 
        for i in range(1, len_dp):
            for j in trues:
                if s[j:i] in wordDict:
                    dp[i] = True
                    trues.append(i)
                    break
        return dp[-1]

        